xesmf/util.py

```diff
@@ -28,6 +28,56 @@ def _grid_1d(start_b, end_b, step):
     return centers, bounds
 
 
+def _cf_bnds_1d(bounds):
+    """Return CF-compliant coordinate bounds.
+
+    Parameters
+    ----------
+    bounds: array (n+1)
+      Array with ordered left and right boundaries.
+
+    Returns
+    -------
+    array (n, 2)
+      Left [:,0] and right [:1] bounds.
+    """
+    return np.vstack((bounds[:-1], bounds[1:])).T
+
+
+def _cf_bnds_2d(bounds):
+    """Return CF-compliant coordinate bounds.
+
+    Parameters
+    ----------
+    bounds: array (n+1, m+1)
+      Array with ordered left and right boundaries.
+
+    Returns
+    -------
+    array (n, m, 4)
+      Counter
+    """
+    return np.dstack((bounds[:-1, :-1], bounds[1:, :-1], bounds[1:, 1:], bounds[:-1, 1:]))
+
+
+def cf_to_esm_bnds_1d(vertices):
+    """Convert 2D CF-compliant bounds to 1D ESM boundaries."""
+    v = vertices
+    return np.concatenate((v[:, 0], v[-1:, 1]))
+
+
+# TODO: This is probably not true in general, and depends on the convention used, starting point, clockwise, etc.
+def cf_to_esm_bnds_2d(vertices):
+    """Convert 3D CF-compliant bounds to 2D ESM boundaries."""
```

Comment:
Here too.

Reply:
I'm pretty sure this implementation is not robust. It assumes coordinates are given starting from the upper left corner and going anti-clockwise.